lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1004,41 +1305,42 @@ private boolean connectImpl(StateIndication request) {
 			this.transport = transport;
 		}
 		if (oldTransport != null) {
-			oldTransport.abort(REASON_TIMEDOUT);
+			oldTransport.close();
 		}
 		transport.connect(this);
 		if(protocolListener != null) {
 			protocolListener.onRawConnectRequested(transport.getURL());
 		}
-		return true;
 	}
 
-	private void closeImpl(StateIndication request) {
-		boolean isConnected = state.state == ConnectionState.connected;
+	/**
+	 * Close any existing transport
+	 * @return closed if true, otherwise awaiting closed indication
+	 */
+	private boolean closeImpl() {
+		if(transport == null) {
+			return true;
+		}
 
-		/* close or abort transport */
-		if(transport != null) {
-			if(isConnected) {
-				/* send a close message on the transport, if connected */
-				try {
-					Log.v(TAG, "Requesting connection close");
-					transport.send(new ProtocolMessage(Action.close));
-				} catch (AblyException e) {
-					transport.abort(e.errorInfo);
-				}
-			} else {
-				/* just close the transport */
-				Log.v(TAG, "Closing incomplete transport");
-				transport.close(false);
-			}
-			transport = null;
+		/* if connected, send an explicit close message and await response */
+		boolean isConnected = currentState.state == ConnectionState.connected;
+		if(isConnected) {
+			try {
+				Log.v(TAG, "Requesting connection close");
+				transport.send(new ProtocolMessage(ProtocolMessage.Action.close));
+				return false;
+			} catch (AblyException e) {}
```

Comment:
This looks a bit suspect. Could you perhaps add a comment here to explain why swallowing exceptions at this point is normal behaviour?

Reply:
https://github.com/ably/ably-java/commit/fdd87ce58cdaff61d0772768086b01d4c35aae95